### ISSUE TITLE
Look into implementing Recognisers.

### ISSUE THREAD

**Issue by SirWumpus** [SirWumpus/post4#57]
> We come to consensus on a number of questions recently, and I just uploaded my [reference implementation](https://github.com/ForthHub/fep-recognizer/tree/master/implementation). I have checked, this implementation works in Post4 too.

@ruv is this the [current  recogniser proposal](https://forth-standard.org/proposals/minimalistic-core-api-for-recognizers) or are there more than one to consider?

**Comment by ruv** [comment]
Yes, this is the current proposal, along with [Common terminology for recognizers](https://forth-standard.org/proposals/common-terminology-for-recognizers-discurse-and-specifications#contribution-161).  At present, the proposal has not yet been updated following recent discussions.

**Comment by SirWumpus** [comment]
> We come to consensus on a number of questions recently, and I just uploaded my [reference implementation](https://github.com/ForthHub/fep-recognizer/tree/master/implementation). I have checked, this implementation works in Post4 too.

@ruv Did you do anything special to test this? I presume you overrode the internal REPL with the recogniser's.

**Comment by ruv** [comment]
> Did you do anything special to test this?

There are tests for recognizers: [name.test.fth](https://github.com/ForthHub/fep-recognizer/blob/master/implementation/variant.gamma/recognizer/name.test.fth), [number.test.fth](https://github.com/ForthHub/fep-recognizer/blob/master/implementation/variant.gamma/recognizer/number.test.fth), [word.test.fth](https://github.com/ForthHub/fep-recognizer/blob/master/implementation/variant.gamma/recognizer/word.test.fth).

They can be run with well-known `ttester.fth`.

> I presume you overrode the internal REPL with the recogniser's.

In this implementation nothing is overridden.  It provides `example.interpret ( i*x "ccc" -- j*x )` and `example.evaluate ( i*x sd -- j*x )` in [example.interpret.fth](https://github.com/ForthHub/fep-recognizer/blob/master/implementation/variant.gamma/example.interpret.fth).

Usage example:
```forth
example.interpret 2 3 + .
s" 2 3 + . " example.evaluate
```
In this code, `2 3 + .`  is interpreted using the recognizer `recognize-forth-default` (that is assigned to the perceptor).

**Comment by SirWumpus** [comment]
SideScuttal:
> They can be run with well-known ttester.fth.

I have my own version `test/assert.p4` that supports the same words and adds a few more.
